indexing-hadoop/src/main/java/io/druid/indexer/path/GranularityPathSpec.java

```diff
@@ -141,8 +143,21 @@ public Job addInputPaths(HadoopDruidIndexerConfig config, Job job) throws IOExce
 
       Path granularPath = new Path(betaInput, intervalPath);
       log.info("Checking path[%s]", granularPath);
-      for (FileStatus status : FSSpideringIterator.spiderIterable(fs, granularPath)) {
-        final Path filePath = status.getPath();
+
+      Iterator<FileStatus> spiderIterator;
+      try {
+        spiderIterator = FSSpideringIterator.spiderIterable(fs, granularPath).iterator();
```

Comment:
I think it would be a lot cleaner for FSSpideringIterator.spiderIterable to return an empty iterator if the file isn't found. This code encodes a lot of assumptions about how FSSpideringIterator.spiderIterable behaves and may be brittle as a result. Like these two things are not really guaranteed:

- File missing exceptions will happen on the call to `iterator` rather than some other point, like on the first call to `hasNext`
- The file missing exception will be the cause of a RuntimeException